Editorial review of EDID to RX Caps (#35)
AMWA-TV/nmos-edid-connection-management: docs/2.0 EDID to Receiver Caps mapping.md

## Background

> Extended Display Identification Data (EDID) is a metadata format for display devices to describe their capabilities to a video source (e.g. graphics card or set-top box). The data format is defined by a standard published by the Video Electronics Standards Association (VESA). This document is targeted against [E-EDID A2][E-EDID] files which consist of EDID 1.4 which is called Base EDID and an optional [CTA-861-G][CTA-861] Extension Block.

Reviewer: ```suggestion
Extended Display Identification Data (EDID) is a metadata format for sink devices to describe their capabilities to a video source (e.g. graphics card or set-top box). The data format is defined by a standard published by the Video Electronics Standards Association (VESA). This document is targeted against [E-EDID A2][E-EDID] files which consist of EDID 1.4 which is called Base EDID and an optional [CTA-861-G][CTA-861] Extension Block.
```

[AMWA BCP-004-01][BCP-004-01] describes a methodology for a similar purpose in relationship to [AMWA IS-04][IS-04].

## Scope

This document provides Best Current Practice guidelines for how to implement a mapping of EDID fields to Receiver Capabilities in order to have a specific way of representing an EDID sink devices through an associated IS-04 Receiver.

## Use of Normative Language

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119][RFC-2119].

## Definitions

_See also the [NMOS Glossary](https://specs.amwa.tv/nmos/branches/main/docs/4.0._Glossary.html), and definitions within RFCs._

### Gateway Device

> A device which encapsulates a digital video or/and audio signal (usually, from a physical connection) coverted into IP packets. Alternatively, the device may perform the operation reversed converting IP packets to a digital signal.

Reviewer: Can we stay with this?
```suggestion
A device which encapsulates digital video or/and audio (usually, from a physical connection) into IP packets or/and decapsulates such signals in the reverse way.
```
`Conversion` looks applicable when talking about similar formats for the same purpose which isn't true for IP packets and an arbitrary digital signal. `En-/Decapsulation` shows what exactly happens with the signal if we talk about SMPTE ST 2022/2110 at least.

Another deal is a necessary comment on decapsulating Gateway Devices.
```suggestion
A device which encapsulates digital video or/and audio (usually, from a physical connection) into IP packets or/and decapsulates such signals in the reverse way. Decapsulating Gateway Devices are physically connected to other Sink Devices.
```

Author: I like your idea about sticking with encapsulation. I am tweaking it to put the two suggestions together.

```suggestion
A device which encapsulates a digital video or/and audio signal (usually, from a physical connection) into IP packets. Conversely, the device may perform the reverse operation for IP packets to a digital signal.
```

> ### Sink - :question: Destination to be consistent with the API

Reviewer: ```suggestion
### Sink
```

A media consuming unit associated with an IS-04 Receiver.

## Video Receivers

If an IS-04 Video Receiver is associated with a Sink which has EDID, its video capabilities MUST be mapped into Receiver's Capabilities accordingly to the rules below.

> If the Sink acts as a Gateway Device, it's supposed to expose its own EDID and therefore Receiver Capabilities and not take into account EDIDs of Sinks connected to the Gateway in a non-networked way. :eyes:

Reviewer: Hi, @prince-chrismc.
I tried to explain the idea in a different way. Could you take a look, please?
```suggestion
If some processing happens between a Receiver and its Sink, (e.g. the Receiver is able to consume 4K Flow and downscale it for the Sink with only Full HD in its EDID), Receiver Capabilities MUST reflect not taking EDID as is. This can also be said for Receivers associated with multiple Sinks at the same time.
```

Author: 👍

> If the Sink acts as a Gateway Device, it's supposed to expose its own EDID instead of EDID of a physical device to take into account any media transformations which the Sink performs. :question:

Reviewer: ```suggestion
If the Sink acts as a Gateway Device, it's supposed to expose its own EDID and therefore Receiver Capabilities and not take into account EDIDs of Sinks connected to the Gateway in a non-networked way.
```

> If some processing happens between a Receiver and its Sink, (e.g. the Receiver is able to consume 4K Flow and downscale it for the Sink with only Full HD in its EDID), Receiver Capabilities MUST reflect not taking EDID as is. This can also be said for Receivers associated with multiple Sinks at the same time.

Reviewer 2: Should we add that "Receiver capabilities" are owned by the Receiver and EDIDs are owned by the Sinks. Thus, "Receiver Capabilities" must always reflect the capabilities of the receiver including conversion?

Author: I think that's a good description of what we are trying to explain

Reviewer: Can we extract this part into Introduction as far as it's common for Video and Audio Receivers?

```
# Introduction
Receiver Capabilities are owned by a Receiver and EDIDs are owned by Sinks. Thus, EDID to Receiver Capabilities mapping MUST be used straightforwardly only if a Receiver passes media to a Sink as is. Otherwise, Receiver Capabilities MUST reflect the capabilities of the Receiver including any types of conversion. This can also be said about Receivers associated with multiple Sinks at the same time.
```

Author: I applied your suggestions!

[RFC-2119]: https://tools.ietf.org/html/rfc2119 "Key words for use in RFCs to Indicate Requirement Levels"

> [IS-04]: https://specs.amwa.tv/is-04/ "AMWA IS-04 NMOS Discovery and Registration Specification (Stable)"

Reviewer: Maybe we need to refer to it somewhere, because it's currently an orphaned link.

Author: it's used here https://github.com/AMWA-TV/nmos-edid-connection-management/pull/35/files#diff-529bd43592af1e685eb0f6f90948951d291dec2ba395423ec709090d04083ae4R7
